test/fuzz/transformation_merge_function_returns_test.cpp

```diff
@@ -0,0 +1,1776 @@
+               OpBranchConditional %11 %38 %37
+         %38 = OpLabel
+         %39 = OpFAdd %7 %35 %35
+               OpReturnValue %39
+         %37 = OpLabel
+               OpReturnValue %35
+               OpFunctionEnd
+)";
+
+  const auto env = SPV_ENV_UNIVERSAL_1_5;
+  const auto consumer = nullptr;
+  const auto context = BuildModule(env, consumer, shader, kFuzzAssembleOption);
+  ASSERT_TRUE(IsValid(env, context.get()));
+
+  spvtools::ValidatorOptions validator_options;
+  TransformationContext transformation_context(
+      MakeUnique<FactManager>(context.get()), validator_options);
+
+  // The 0s are allowed because the function's return type is void.
+  auto transformation1 =
+      TransformationMergeFunctionReturns(14, 100, 101, 0, 0, {{}});
+  ASSERT_TRUE(
+      transformation1.IsApplicable(context.get(), transformation_context));
+  transformation1.Apply(context.get(), &transformation_context);
+  ASSERT_TRUE(IsValid(env, context.get()));
+
+  // %12 is available at the end of the entry block of %19 (it is a global
+  // variable).
+  ASSERT_TRUE(TransformationMergeFunctionReturns(19, 110, 111, 112, 12, {{}})
+                  .IsApplicable(context.get(), transformation_context));
+
+  // %1000 cannot be found in the module, but there is a suitable id available
+  // at the end of the entry block (%12).
+  auto transformation2 =
+      TransformationMergeFunctionReturns(19, 110, 111, 112, 1000, {{}});
+  ASSERT_TRUE(
+      transformation2.IsApplicable(context.get(), transformation_context));
+  transformation2.Apply(context.get(), &transformation_context);
+  ASSERT_TRUE(IsValid(env, context.get()));
+
+  // %27 is available at the end of the entry block of %26 (it is a function
+  // parameter).
+  ASSERT_TRUE(TransformationMergeFunctionReturns(26, 120, 121, 122, 27, {{}})
+                  .IsApplicable(context.get(), transformation_context));
+
+  // %1000 cannot be found in the module, but there is a suitable id available
+  // at the end of the entry block (%27).
+  auto transformation3 =
+      TransformationMergeFunctionReturns(26, 120, 121, 122, 1000, {{}});
+  ASSERT_TRUE(
+      transformation3.IsApplicable(context.get(), transformation_context));
+  transformation3.Apply(context.get(), &transformation_context);
+  ASSERT_TRUE(IsValid(env, context.get()));
+
+  // %35 is available at the end of the entry block of %33 (it is in the entry
+  // block).
+  ASSERT_TRUE(TransformationMergeFunctionReturns(26, 130, 131, 132, 27, {{}})
+                  .IsApplicable(context.get(), transformation_context));
+
+  // %1000 cannot be found in the module, but there is a suitable id available
+  // at the end of the entry block (%35).
+  auto transformation4 =
+      TransformationMergeFunctionReturns(33, 130, 131, 132, 1000, {{}});
+  ASSERT_TRUE(
+      transformation4.IsApplicable(context.get(), transformation_context));
+  transformation4.Apply(context.get(), &transformation_context);
```

Comment:
Can you please rebase, and then change the tests so that they do not call "Apply" directly?  (See what the other tests in the project now do.)  You will also need to implement GetFreshIds() in the transformation - should be obvious based on what other transformations do.

Reply:
Done